File: AmadeusBack/amadeus-api/database/models/TodoTask.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Security.Policy;

namespace amadeus_api.database.models;

public class TodoTask : Entity
{
    [ForeignKey(nameof(Owner))]
    public long OwnerId { get; set; }
    public User Owner { get; set; } = null!;

    public TodoTaskStatus Status { get; set; } = TodoTaskStatus.TODO;
    public string? Description { get; set; }
    public DateTime? DueDate { get; set; }

    public override void MarkDeleted(long? userId)
    {
        base.MarkDeleted(userId);
        Status = TodoTaskStatus.DELETED;
    }
}

File: AmadeusFront/src/app/models/TodoTask.ts
import { Entity } from "./Entity";

export interface TodoTask extends Entity {
    description: string | null;
    status: string;
    dueDate: Date | null;
}

export enum TodoTaskStatus {
    TODO = "TODO",
    IN_PROGRESS = "IN_PROGRESS",
    DONE = "DONE",
    DELETED = "DELETED"
}